source/Cosmos.System2/Graphics/Canvas.cs

```diff
@@ -641,9 +641,37 @@ public void DrawImageAlpha(Image image, int x, int y)
                 for (int yi = 0; yi < image.Height; yi++)
                 {
                     color = Color.FromArgb(image.RawData[xi + (yi * image.Width)]);
-                    DrawPoint(color, x + xi, y + yi);
+                    if (color.A >0)
+                    {
+                        DrawPoint(color, x + xi, y + yi);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets a rectangle of pixels and stores it into a bitmap
+        /// </summary>
+        /// <param name="X">the x coordinate of the rectangle</param>
+        /// <param name="Y">the y coordinate of the rectangle</param>
+        /// <param name="W">the width of the rectangle</param>
+        /// <param name="H">the height of the rectangle</param>
+        /// <returns>bitmap</returns>
+        public Bitmap GetPixels(int X,int Y,ushort W,ushort H)
```

Comment:
i dont know about this i feel it does not belong in canvas

Reply:
we already have the GetPoint function it's just a quality of life feature

Reply:
@zarlo where should i move it

Reply:
We could move it to the Image as a new constructor